bin/python/combine_reports_from_date_range.py

```diff
@@ -0,0 +1,109 @@
+"""Script for combining reports"""
+import os
+import sys
+import datetime
+import re
+import json
+import functools
+
+import logger
+import configurator
+from gcp import GcpManager
+from report import MdTable
+
+LOGGER = logger.get_logger('combine_reports_from_date_range')
+def _reduce_labels(sets):
+    return sorted(functools.reduce(lambda cur, acc: acc.union(cur), sets, set()))
+
+def _render_results(results):
+    result_label_sets = map(lambda test: set(test.keys()), results['tests'].values())
+    result_labels = _reduce_labels(result_label_sets)
+    tests_table = MdTable(['test'] + result_labels)
+    if results['tests']:
+        for test, result in results['tests'].items():
+            tests_table.add_row([test, *[str(result.get(label, 0)) for label in result_labels]])
+    else:
+        tests_table.add_row(['No results'])
+
+    category_label_sets = map(lambda category: set(category.keys()), results['categories'].values())
+    result_labels = _reduce_labels(category_label_sets)
+    categories_table = MdTable(['categories'] + result_labels)
+    if results['categories']:
+        for category, result in results['categories'].items():
+            row_result = [str(result.get(label, 0)) for label in result_labels]
+            categories_table.add_row([category, *row_result])
+    else:
+        categories_table.add_row(['No results'])
+
+    missing_table = MdTable(['missing tests', 'count'])
+    if results['missing']:
+        for test in sorted(results['missing'].keys()):
+            missing_table.add_row([test, str(results['missing'][test])])
+    else:
+        missing_table.add_row(['None'])
+
+    return "\n%s\n" * 3 % (tests_table.render(), categories_table.render(), missing_table.render())
+
+def _get_local_reports(device, reports_dir, start, end):
+    LOGGER.info('Looking for reports locally')
+    prog = re.compile(r'^report_%s_\d{4}-\d{2}-\d{2}T\d{6}\+\d{4}.*\.json$' % device)
+    ts = re.compile(r'\d{4}-\d{2}-\d{2}T\d{6}\+\d{4}')
```

Comment:
short names generally aren't allowed (except for things like 'e') -- timestamp?  or ts_match?